components/server/src/server.rs

```diff
@@ -1218,38 +1207,76 @@ impl<ER: RaftEngine> TiKVServer<ER> {
     }
 }
 
-impl TiKVServer<RocksEngine> {
-    fn init_raw_engines(
-        &mut self,
-        flow_listener: engine_rocks::FlowListener,
-    ) -> (Engines<RocksEngine, RocksEngine>, Arc<EnginesResourceInfo>) {
-        let env = get_env(self.encryption_key_manager.clone(), get_io_rate_limiter()).unwrap();
-        let block_cache = self.config.storage.block_cache.build_shared_cache();
+/// This trait handles setup steps that differ between implementations of
+/// KvEngine. As alternate engines reach feature parity with RocksEngine, it
+/// should be possible to remove methods or even the whole trait.
+trait CreateKvEngine<ER>: KvEngine
+where
+    ER: RaftEngine,
+{
+    fn create_kv_engine(
+        config: &TiKvConfig,
+        region_info_accessor: &RegionInfoAccessor,
+        store_path: &Path,
+        router: &RaftRouter<Self, ER>,
+        env: Arc<engine_rocks::raw::Env>,
```

Comment:
it's still specific to rocksDB

Reply:
Yes, that's right. There are several features that remain rocks-specific for now, including the debug and backup services and this encryption configuration. This patch enables non-rocks engines to ignore those features and still boot.

Reply:
It's not in this series, but a patch that introduces other engines implements some of these CreateKvEngine methods as no-ops.